src/backends/NFTInterface.ts

```diff
@@ -0,0 +1,124 @@
+import { KNOWN_CONTRACTS } from '../fetcher/FetchResultTypes';
+import { ETHAddress } from '../types/standard';
+
+type Nullable<T> = T | null;
+
+export type CurrencyValue = {
+  usdValue?: string;
+  ethValue?: string;
+  symbol: string;
+  name?: string;
+  currency: string;
+  decimals?: number;
+  amount: string;
+  prettyAmount: string;
+};
+
+export type TimedAction = {
+  timestamp: number;
+  blockNumber: Nullable<number>;
+  transactionHash: Nullable<string>;
+};
+
+export type AuctionBidEvent = {
+  creator: string;
+  amount: CurrencyValue;
+  created: TimedAction;
+};
+
+export type AuctionLike = {
+  winner?: string;
+  endsAt?: TimedAction;
+  duration: number;
+  startedAt?: TimedAction;
+  currentBid?: AuctionBidEvent;
+  // current bid is duplicated within bids
+  bids: AuctionBidEvent[];
+  source: 'ZoraReserveV0' | 'OpenseaEnglish';
+  type: 'Auction';
+} & MarketInfo;
+
+export type FixedPriceLike = {
+  side: 'ask' | 'offer';
+  expires?: number;
+  source: 'ZNFTPerpetual' | 'ZoraAskV1' | 'OpenseaFixed';
+  type: 'FixedPrice';
+} & MarketInfo;
+
+type MarketInfo = {
+  raw: any;
+  amount: CurrencyValue;
+  // pending - inactive pending some event
+  // active - can be filled / auction is ongoing
+  // completed - auction end fill complete
+  // cancelled - user cancels at some point
+  status: 'pending' | 'active' | 'complete' | 'cancelled' | 'unknown';
+  createdAt: TimedAction;
+  createdBy?: string;
+  finishedAt?: TimedAction;
+  cancelledAt?: TimedAction;
+};
+
+export type MarketModule = AuctionLike | FixedPriceLike;
+
+export type MediaObject = {
+  uri: string;
+  mime?: string;
+};
+
+export type MetadataAttributeType = {
```

Comment:
this type doesn't actually match the response yet, assuming WIP, might be worth leaving or something with a way of using defaults 

```
interface BaseMetadataAttributeType {
  trait_type: string
}

interface FormattedMetadataAttributeType extends BaseMetadataAttributeType {
...
}

``` 

just so the type is valid / we can only show the attributes if its valid maybe.